sympy/diffgeom/diffgeom.py

```diff
@@ -1084,6 +1086,388 @@ def __call__(self, field):
     def _latex(self, printer, *args):
         return r'\mathbb{\nabla}_{%s}' % printer._print(self._wrt)
 
+##########################################################################
+# Tensors as multidimensional arrays.
+##########################################################################
+
+class TensorArray(object):
```

Comment:
If this class is mutable, maybe it should be called `MutableTensorArray`. In SymPy objects are usually immutable, unless marked as mutable.